### ISSUE TITLE
Panic: Preview can't handle some chars when filetype is txt?

### ISSUE THREAD

**Issue by Enayaaa** [kyoheiu/felix#169]
Running `fx` with previewing on, I tried to scroll to a file having the following content:
```
████████████████████████████████
```
it seems that the filetype is also important, as this error only occured if my filetype was .txt as far as my testing showed.

`fx` panicks at such file content and prints the following.
```
Panic: byte index 94 is not a char boundary; it is inside '█' (bytes 93..96) of `████████████████████████████████
`
Error: felix panicked
```
The problem does not occur if I have previewing off (pressing `v` while `fx` is open somewhere).

To reproduce: place a file with the above content and .txt fileending in an empty directory and try to run `fx` in the directory.

**Comment by kyoheiu** [comment]
Could you tell me your felix version (‘felix --version‘) and OS?

**Comment by Enayaaa** [comment]
felix v2.2.2
Linux pop-os 6.0.12-76060006-generic #202212290932~1671652965~22.04~452ea9d SMP PREEMPT_DYNAMIC Wed D x86_64 x86_64 x86_64 GNU/Linux

**Comment by kyoheiu** [comment]
Hm, I tried to reproduce this. but couldn't.

> To reproduce: place a file with the above content and .txt fileending in an empty directory and try to run fx in the directory.

![Peek 2023-01-11 09-25](https://user-images.githubusercontent.com/61998590/211689768-8b4b67bf-ba66-42d9-bc78-c1d82e62315e.gif)

As shown above, with the text you gave I created a `.txt` file and executed `fx` in the directory. Preview can show the text (although the positions of new lines are not correct, which is due to the need to handle the multibyte text as properly as this app can for now.)

Basically this `char_boundary`-related issue should have been fixed by the commit 7b1d5bf0565dedadc48a1d42cc2d22c28825dad5 .

**Comment by kyoheiu** [comment]
FYI, felix detects file type by `content_inspector::inspect`, so the extension is not relevant.
After that, the text content is split by:
1. native function (when syntax-highlighting is off) or
2. `syntect::util::split_at` (my fork version).
So it may be useful to know whether the highlighting is on.

**Comment by kyoheiu** [comment]
Happy to hear that the panic does not occur with this repository: As I understand, I should've set `[patch]` to tell crates.io to use my own fork version of `syntect`, not only specifying the url in the dependencies.
(The panic is due to the latest version of `syntect` panicking when splitting texts with multibyte chars: I created PR for this about a months ago)

Even when you install this app via cargo, if the syntax-highlighting is off the panic does not occur I think - because the syntect panic happens only if the highlighting is on.

In the next pubishing to the crate.io I'll fix this. Thank you for the report!

**Comment by kyoheiu** [comment]
v2.2.3 released, but this issue is still alive with `cargo install felix` due to `syntect` not updating yet.
FYI, v2.2.3 fixes wide chars handling and the preview feature. If you'd like to use this latest version, please install from this repo.

**Comment by kyoheiu** [comment]
Updating syntect solved this.
